samples: watchdog: Update watchdog example to new API

kl-cruz opened the pull request:
Updating watchdog sample application to new watchdog API. Fixes https://github.com/zephyrproject-rtos/zephyr/issues/8215

kl-cruz committed:
samples: watchdog: Update watchdog example to new API

Watchdog example is updated to use new watchdog API.

Signed-off-by: Karol Lasończyk <karol.lasonczyk@nordicsemi.no>

anangl commented on samples/drivers/watchdog/prj.conf:
```diff
@@ -1 +1,2 @@
 CONFIG_WATCHDOG=y
+CONFIG_WDT_NRFX=y
```
It does not seem to be a good idea to make this sample "Nordic only".

kl-cruz replied:
It would require to enable Nordic's watchdog in the other way

anangl replied:
I am not sure I understand. Is it intentional to make this sample to build only for Nordic chips?
The `WDT_NRFX` will not be available for other chips, since it depends on `SOC_FAMILY_NRF`, so cmake will complain.

anangl replied:
Alternatively, you can add `default y` to the `WDT_NRFX` definition. Then this option will be automatically activated when the watchdog is enabled and the build is done for a Nordic SoC.

kl-cruz replied:
Yes, that's it. This definition has to be set to ` y ` somewhere. Your idea with setting ` default y ` looks good. I'll update test's ` proj.conf ` file too.

kl-cruz committed:
tests/samples: watchdog: Update projects' configuration

WDT_NRFX is now enabling by default. Its configuration from proj.conf
files was removed.

Signed-off-by: Karol Lasończyk <karol.lasonczyk@nordicsemi.no>
